src/admin/index.js

```diff
@@ -620,6 +619,73 @@ module.exports = ({
     return { groups }
   }
 
+  /**
+   * List groups in a broker
+   *
+   * @param {string[]} [groupIds]
+   * @return {Promise<DeleteGroups>}
+   *
+   * @typedef {Object} DeleteGroups
+   * @property {Array<DeleteGroup>} results
+   *
+   * @typedef {Object} DeleteGroup
+   * @property {string} groupId
+   * @property {number} errorCode
+   */
+  const deleteGroups = async groupIds => {
+    if (!groupIds || !Array.isArray(groupIds)) {
+      throw new KafkaJSNonRetriableError(`Invalid groupIds array ${groupIds}`)
+    }
+
+    const invalidGroupId = groupIds.some(g => typeof g !== 'string')
+
+    if (invalidGroupId) {
+      throw new KafkaJSNonRetriableError(`Invalid groupId name: ${JSON.stringify(invalidGroupId)}`)
+    }
+
+    const retrier = createRetry(retry)
+
+    return retrier(async (bail, retryCount, retryTime) => {
```

Comment:
This is going to be a little tricky, but you'll need to store the responses outside of the retrier and only retry those requests that don't have a valid result yet.

Imagine you're trying to delete 3 groups: a, b, c.

On the first try, you send off 3 requests, a and b work but c fails, so the promise rejects.

You retry, making the same 3 requests again. c works this time, but you are still trying to delete a and b again. At this point I'm not sure what will happen, but I wouldn't be surprised if you get back a `GROUP_ID_NOT_FOUND` error for a and b, which means that again you are going to retry deleting a, b and c.

Instead, you'll need to wait for all requests to either reject or resolve, and then only retry those ones that rejected. Meaning that in the same scenario, you first make requests for a, b, c. a and b resolve and c rejects, so you retry, but since you already have responses for a and b, you only make a request for c.

We have the same problem somewhere else, where we solved it like that, but I can't remember where. I thought it was in `ConsumerGroup.fetch`, but it doesn't look like it, looking at the code now. Maybe @tulios remembers?

Reply:
Did it now in a way that it changes given groupIds on retries